lib/JsonStreamQueryTransformer.ts

```diff
@@ -0,0 +1,137 @@
+import {Transform} from "readable-stream";
+import {JsonEvent} from "json-event-parser";
+
+export type JsonValue = string | number | boolean | null | JsonValue[] | {[key: string]: JsonValue};
+
+export type QueryDefinition = {
+  id: string;
+  path: (string | null)[],
+  returnContent: boolean
+}
+
+export type QueryResult = {
+  query: string;
+  value?: JsonValue
+}
+
+/**
+ * Emits all JSON blobs matching one of the given query.
+ * A query is a path composed of object keys or `null` for array keys.
+ */
+export class JsonStreamQueryTransformer extends Transform {
```

Comment:
If I understand correctly, this is an abstraction layer on top of json-event-parser.js, but is independent of SPARQL/JSON.
If so, would it make sense to add this to json-event-parser.js (with the necessary tests)?
I suspect this abstraction layer could be highly valuable for other streaming JSON usages.

Reply:
>  If so, would it make sense to add this to json-event-parser.js (with the necessary tests)?
>  I suspect this abstraction layer could be highly valuable for other streaming JSON usages.

That's a great question. Yes, indeed it might be valuable. I am a bit unhappy with current API state that is tailored to SPARQL/JSON needs. Maybe implementing JSONPath could be handy.

Reply:
JSONPath would probably require in quite a bit more effort, but it's up to you of course. I would already be very happy  with having just this in json-event-parser.js :-)

Reply:
Done: https://github.com/comunica/json-event-parser.js/pull/1